### PR TITLE
Celestia  Compression PR-2: Envelope and BlobWithSender plumbing

### PR DESCRIPTION
**Pull request by citizen-stig**
# Description

**Celestia Compression PR-2: Envelope format + `BlobWithSender` plumbing.**

Second PR in the Celestia blob-compression series, stacked on PR-1 (#2977). This is **plumbing only — no change to live blob submit/read/verify behavior.** It introduces the compressed-blob envelope format and the cache slot that PR-3 will wire into the read path.

> Note: the branch is named `…witness-pruning`, but witness pruning is explicitly out of scope for this series (see the design doc). The actual content is the envelope format + `BlobWithSender` plumbing.

## What's in it

1. **New `envelope` module** (`crates/adapters/celestia/src/envelope.rs`) — pure, deterministic, *ungated* logic that recognizes and validates the v1 envelope from a blob's authenticated DA-physical bytes. Because it is side-effect-free and ungated, the **same compiled code runs in native extraction and the guest verifier**, so identical bytes classify identically.
   - Fixed **24-byte header**: `magic[16] = "SOV_CELESTIA_CMP"`, `version=1`, `codec` (0 = raw escape, 1 = LZ4, 2 reserved/zstd), `flags=0`, `logical_len` (u32 LE).
   - `parse_envelope()` / `classify()` → `EnvelopeState::{Legacy, Malformed, Envelope(header)}`. No-magic ⇒ `Legacy`; magic-but-invalid ⇒ `Malformed` (treated as authenticated raw bytes, i.e. slashable). No decode happens here — PR-2 only validates the header.
   - **Security caps checked before any decode**: `MAX_LOGICAL_BLOB_LEN` (64 MiB) bounds allocation; `MAX_COMPRESSION_RATIO` (64×) rejects decompression bombs (multiply-form to avoid div-by-zero on empty payloads); raw-escape requires `payload_len == logical_len`.

2. **`BlobWithSender` envelope cache** (`crates/adapters/celestia/src/types/mod.rs`) — a `#[serde(skip)] OnceLock<EnvelopeState>` interior cache derived **only** from the authenticated DA-physical accumulator (never a serialized witness claim). Empty after construction and after every (de)serialization, then recomputed deterministically. A hand-written `PartialEq` ignores the cache (derived state ≠ identity). PR-3 fills it lazily via `get_or_init` holding only `&self`.

3. **Design doc** `docs/design/compression_actual_plan.md` (v7) — records the format and the key decisions (full-frame not chunked; runtime config controls *emission* only; proof math stays on physical bytes; no fabricated logical bytes).

## What reviewers should focus on

- **Wire format + the two caps** (`MAX_LOGICAL_BLOB_LEN`, `MAX_COMPRESSION_RATIO`) — these become consensus-relevant invariants once PR-3 reads envelopes.
- **Native/guest determinism**: confirm the module stays ungated and side-effect-free (the `classification_table_is_fixed` test documents this guarantee).
- **Cache semantics**: serde-skip + custom `PartialEq` + `OnceLock`, and that classification derives from authenticated bytes only.
- **No behavior change**: `#[allow(dead_code)]` confirms nothing is wired in yet.

The clean review diff is **5 files** (against the PR-1 base branch): the new `envelope.rs`, the `BlobWithSender` change in `types/mod.rs`, the module declaration in `lib.rs`, the design doc, and `CHANGELOG.md`.

- [ ] I have updated `CHANGELOG.md` with a new entry if my PR makes any breaking changes or fixes a bug. If my PR removes a feature or changes its behavior, I provide help for users on how to migrate to the new behavior.
- [ ] I have carefully reviewed all my `Cargo.toml` changes before opening the PRs. (Are all new dependencies necessary? Is any module dependency leaked into the full-node (hint: it shouldn't)?)

## Linked Issues
- Related to #2990

## Testing
- Unit tests in `envelope.rs`: endianness, every malformed/error variant (bad version, nonzero flags, unsupported codec, both caps, length mismatch, too-short header), zero-length edge cases incl. the div-by-zero guard, and a fixed-classification determinism table.
- `serde_roundtrip_drops_envelope_cache_and_recomputes` in `types/mod.rs`: proves the cache is dropped on (de)serialize, equality ignores it, and reconstruction from the authenticated bytes is deterministic.

## Docs
Format and design decisions are documented in `docs/design/compression_actual_plan.md` (v7). Rendered docs are available at `https://sovlabs-ci-rustdoc-artifacts.us-east-1.amazonaws.com/<BRANCH_NAME>/index.html`.

### CODE REVIEW

File: crates/adapters/celestia/src/envelope.rs [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+/// that the header advertises a known codec.
+pub(crate) const CODEC_LZ4: u8 = 1;
+
+/// Fixed header size: magic[16] + version[1] + codec[1] + flags[2] + logical_len[4].
+pub(crate) const ENVELOPE_HEADER_LEN: usize = 24;
+
+/// Hard cap on the decoded logical payload length.
+///
+/// 64 MiB sits comfortably above any realistic rollup batch (the per-tx
+/// `MAX_TX_SIZE` ceiling is 8 MiB) yet well below the ~126 MiB a Celestia data
```
Comment (citizen-stig) [inline crates/adapters/celestia/src/envelope.rs:46]: todo: celestia uncompressed square size is not related.
